src/generators/php/presets/JsonSerializablePreset.ts

```diff
@@ -0,0 +1,88 @@
+import { PhpPreset } from '../PhpPreset';
+import { PhpRenderer } from '../PhpRenderer';
+import { ConstrainedMetaModel } from '../../../models';
+
+function renderSelf({
+  content,
+  renderer
+}: {
+  content: string;
+  renderer: PhpRenderer<ConstrainedMetaModel>;
+}): string {
+  renderer.dependencyManager.addDependency('use JsonSerializable;');
+
+  const contentLines = content.split('\n');
+  contentLines[0] += ` implements JsonSerializable`;
+
+  return contentLines.join('\n');
+}
+
+/**
+ * Preset, which implements PHP’s JsonSerializable interface.
+ *
+ * Using this will allow to json serialize the model using `json_encode()`.
+ *
+ * @implements {PhpPreset}
+ */
+export const PHP_JSON_SERIALIZABLE_PRESET: PhpPreset = {
+  class: {
+    self({ content, renderer }): string {
+      return renderSelf({ content, renderer });
+    },
+    additionalContent({ renderer, model, content }): string {
+      return (
+        content +
+        renderer.renderBlock([
+          'public function jsonSerialize(): array',
+          '{',
+          renderer.indent(
+            renderer.renderBlock([
+              'return [',
+              renderer.indent(
+                renderer.renderBlock(
+                  Object.values(model.properties).map((property) => {
+                    if (property.propertyName === 'additionalProperties') {
+                      return `...$this->${property.propertyName},`;
+                    }
+
+                    return `'${property.unconstrainedPropertyName}' => $this->${property.propertyName},`;
+                  })
+                )
+              ),
+              '];'
+            ])
+          ),
+          '}'
+        ])
+      );
```

Comment:
All these nested renderBlock and indent calls can quickly become unmaintainable. 

An alternative is to do something along the following:
```suggestion
  const serializedProperties = Object.values(model.properties).map((property) => {
    if (property.property instanceof ConstrainedDictionaryModel && property.property.serializationType === 'unwrap') {
      return `...$this->${property.propertyName}`;
    }
  
    return `'${property.unconstrainedPropertyName}' => $this->${property.propertyName}`;
  });

  return `${content}
  public function jsonSerialize(): array
  {
    return [
     ${renderer.indent(serializedProperties.join(',\n'))}
    ];
  }
```

Reply:
The indention is configurable. The suggested solution will not make use of the defined size and type of indention.

I applied the suggestion only partially in f2616f9d2e741282ccee785ecefea35b9c6f5299